entities.py

```diff
@@ -0,0 +1,221 @@
+"""
+LibrePCB S-expression entities
+"""
+
+from common import format_float
+from typing import List
+
+
+class DateValue(object):
+    """Helper class to represent a single named date value"""
+    def __init__(self, name: str, date: str):
+        self.name = name
+        self.date = date
+
+    def __str__(self) -> str:
+        return '({} {})'.format(self.name, self.date)
+
+
+class UUIDValue(object):
+    """Helper class to represent a single named UUID value"""
+    def __init__(self, name: str, uuid: str):
+        self.name = name
+        self.uuid = uuid
+
+    def __str__(self) -> str:
+        return '({} {})'.format(self.name, self.uuid)
+
+
+class BoolValue(object):
+    """Helper class to represent a single named boolean value"""
+    def __init__(self, name: str, value: bool):
+        self.name = name
+        self.value = str(value).lower()
+
+    def __str__(self) -> str:
+        return '({} {})'.format(self.name, self.value)
+
+
+class StringValue(object):
+    """Helper class to represent a single named string value"""
+    def __init__(self, name: str, value: str):
+        self.name = name
+        self.value = value
+
+    def __str__(self) -> str:
+        return '({} "{}")'.format(self.name, self.value)
+
+
+class FloatValue(object):
+    """Helper class to represent a single named float value"""
+    def __init__(self, name: str, value: float):
+        self.name = name
+        self.value = value
+
+    def __str__(self) -> str:
+        return '({} {})'.format(self.name, format_float(self.value))
+
+
+class Name(StringValue):
+    def __init__(self, name: str):
+        super().__init__('name', name)
```

Comment:
These classes can be simplified using partial application:

```python
from functools import partial

Name = partial(StringValue, 'name')
Description = partial(StringValue, 'description')
...
```

Reply:
Hmm I'm not sure if I like this if we think about adding documentation to these classes later on:
```python
>>> from functools import partial
>>> basetwo = partial(int, base=2)
>>> basetwo.__doc__ = 'Convert base 2 string to an int.'
```

Also it doesn't play well with type safety:
```
In [1]: from functools import partial                                                                                                                          

In [2]: class Foo: 
   ...:     def __init__(self, a, b): 
   ...:         self.a = a 
   ...:         self.b = b 
   ...:                                                                                                                                                        

In [3]: Bar = partial(Foo, 1)                                                                                                                                  

In [4]: b = Bar(2)                                                                                                                                             

In [5]: type(b)                                                                                                                                                
Out[5]: __main__.Foo
```
So it is still the base type which is not correct.

Reply:
Yeah, `partial` is just partial application, it's not a real subtype :) Use whatever you prefer.

Classes might be better because they can be used as type in mypy.